File: android/app/src/main/java/com/iniyan/copypaste/CopiedData.kt
package com.iniyan.copypaste

import com.google.firebase.Timestamp

class CopiedData constructor(
    private val data: String,
    private val device: String,
    private val time: Timestamp
) {
    companion object {
        fun fromJson(map: Map<String, Any>): CopiedData {
            return CopiedData(
                data = map["data"] as String,
                device = map["device"] as String,
                time = map["time"] as Timestamp
            )
        }
    }


    fun toJson(): Map<String, Any> {
        return mapOf<String, Any>(
            "data" to data,
            "device" to device,
            "time" to time
        )
    }
}

File: lib/constants/copied_data.dart
class CopiedData {
  CopiedData({
    required this.data,
    required this.device,
    required this.time,
  });

  String device;
  String data;
  DateTime time;

  Map<String, dynamic> toJson() => {
        "data": data,
        "device": device,
        "time": time,
      };

  static CopiedData fromJson(Map<String, dynamic> json) => CopiedData(
        data: json['data'],
        device: json['device'],
        time: json['time'].toDate(),
      );
}
